oshmem/mca/spml/ucx/spml_ucx.c

```diff
@@ -1676,4 +1692,118 @@ int mca_spml_ucx_team_reduce(shmem_team_t team, void
     return OSHMEM_ERR_NOT_IMPLEMENTED;
 }
 
+int mca_spml_ucx_rkey_store_create(mca_spml_ucx_rkey_store_t *store,
+                                   ucp_context_h ucp_context,
+                                   int size)
+{
+    store->array = calloc(size, sizeof(*store->array));
+    if (store->array == NULL) {
+        return OSHMEM_ERR_OUT_OF_RESOURCE;
+    }
+
+    store->size  = size;
+    store->count = 0;
+    store->ucp_context = ucp_context;
+    return OSHMEM_SUCCESS;
+}
+
+void mca_spml_ucx_rkey_store_destroy(mca_spml_ucx_rkey_store_t *store)
+{
+    int i;
 
+    for (i = 0; i < store->count; i++) {
+        assert(store->array[i].refcnt == 0);
+        ucp_rkey_destroy(store->array[i].rkey);
+    }
+
+    free(store->array);
+    store->count = 0;
+    store->size  = 0;
+}
+
+static int mca_spml_ucx_rkey_store_find(const mca_spml_ucx_rkey_store_t *store,
+                                        const ucp_rkey_h target,
+                                        int *index)
+{
+    ucp_rkey_compare_params_t params = {};
+    int i, end, result, m;
+    mca_spml_ucx_rkey_t *entry;
+    ucs_status_t status;
+
+    for (i = 0, end = store->count; i < end;) {
+        m = (i + end) / 2;
```

Comment:
maybe use "bsearch" from glibc?

Reply:
`bsearch` returns `NULL` when not found, but here instead we return the position where the item can be inserted (`OSHMEM_ERR_NOT_FOUND`), so i think is simpler as it is, wdyt?